src/main/java/roomescape/reservation/ReservationService.java

```diff
@@ -0,0 +1,70 @@
+package roomescape.reservation;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+
+    public ReservationService(ReservationRepository reservationRepository) {
+        this.reservationRepository = reservationRepository;
+    }
+
+    public List<ReservationResponseDto> findAll() {
+        List<Reservation> reservations = reservationRepository.findAll();
+        return reservations.stream()
+                .map(
+                        reservation -> new ReservationResponseDto.Builder()
+                                .id(reservation.getId())
+                                .name(reservation.getName())
+                                .date(reservation.getDate())
+                                .time(reservation.getTime())
+                                .build()
+                ).toList();
+    }
+
+    @Transactional
+    public ReservationResponseDto save(ReservationRequestDto reservationRequestDto) {
+        final Reservation reservation = new Reservation.Builder().name(reservationRequestDto.getName())
+                .date(reservationRequestDto.getDate())
+                .time(reservationRequestDto.getTime())
+                .build();
+
+        final Long savedId = reservationRepository.save(reservation);
+        final Reservation savedReservation = reservationRepository.findById(savedId);
+
+        return new ReservationResponseDto.Builder()
+                .id(savedReservation.getId())
+                .name(savedReservation.getName())
+                .date(savedReservation.getDate())
+                .time(savedReservation.getTime())
+                .build();
+    }
+
+
+    @Transactional
+    public void delete(Long id) {
+        if(!reservationRepository.existsById(id)) {
+            throw new IllegalArgumentException("해당 예약이 존재하지 않습니다.");
+        } else if (reservationRepository.existsById(id)) {
+            reservationRepository.deleteById(id);
+        }
+    }
```

Comment:
삭제를 수행할 때 쿼리가 총 3회 수행되겠네요!
질의를 한번 한 후, 삭제가 수행되도록 하여 총 2회만 쿼리가 실행되도록 하면 좋겠어요!

```suggestion
    @Transactional
    public void delete(Long id) {
        final boolean isExistedReservation = reservationRepository.existsById(id);
        if (isExistedReservation) {
            reservationRepository.deleteById(id);
        }
        throw new IllegalArgumentException("해당 예약이 존재하지 않습니다.");
    }
```